packages/gatsby-theme/src/components/Share/Share.js

```diff
@@ -1,25 +1,52 @@
 import React from "react"
 import PropTypes from "prop-types"
 import { Box } from "theme-ui"
+import { Heading } from "ui"
 
-const Share = ({ children, ...props }) => {
+const Share = ({ children, heading, sx, ...props }) => {
   return (
     <Box
       __css={{
         py: "small",
         display: "flex",
         flexWrap: "wrap",
         alignItems: "center",
+        ...sx,
       }}
       {...props}
     >
-      {children}
+      <Heading
+        sx={{
+          color: "secondary",
+          m: 0,
+          mr: "small",
+        }}
+      >
+        {heading}
+      </Heading>
```

Comment:
Heading could be undefined and this will make an empty `h2`, lets validate if there is a heading or not
```suggestion
      {heading & 
      <Heading
        sx={{
          color: "secondary",
          m: 0,
          mr: "small",
        }}
      >
        {heading}
      </Heading>}
```